scripts/service-catalog.py

```diff
@@ -0,0 +1,270 @@
+
+    return portfolio_id
+
+def main(temp_s3_url,product_name,conn,product_template,portfolio_id):
+    ser_cat_clt_conn = conn[0]
+    region = conn[1]
+    client_s3 = conn[2]
+
+    """TO create product
+    """
+    response = ser_cat_clt_conn.search_products()
+    for product in response['ProductViewSummaries']:
+        if product['Name'] == product_name:
+            product_id =  product['ProductId']
+            version_response = ser_cat_clt_conn.describe_product(Id=product_id)
+            tdict= {}
+            vdict= {}
+            for version in  version_response['ProvisioningArtifacts']:
+                tdict[time.mktime(version['CreatedTime'].timetuple())]=version['Id']
+                vdict[time.mktime(version['CreatedTime'].timetuple())]=version['Name']
+
+            latest_version_id= tdict[max(tdict.keys())]
+            latest_version_name= vdict[max(vdict.keys())]
+            print latest_version_id
+            print latest_version_name
+            template_utl = get_latest_version_template(ser_cat_clt_conn,latest_version_id,product_id)
+            print "product_template=",product_template
+
+            comp_status = compare_templates(conn,template_utl,product_name,product_template)
+
+            # create version of product if template changed
+            if comp_status[0] == True:
+                global VERSION
+                VERSION = "v"+str(float(latest_version_name.split("v")[1])+.1)
+                print VERSION
+                # upload new template to bucket for new version
+                template_info = put_template_in_s3(client_s3,comp_status[1])
+                create_version_of_product(ser_cat_clt_conn,VERSION,template_info[1],product_id,product_name,region,client_s3)
+
+            break
+
+            """
+            for version in version_response['ProvisioningArtifacts']:
+                if "v1.0.0" == version['Name']:
+                    print "product {} with version {}   already exist in region {} and attached with portfolio {}".format(product_name,"v1.0.0",region,PORTFOLIO_NAME)
+                    break
+            else:
+                #To create New  Version of  existing product
+                create_version_of_product(ser_cat_clt_conn,"v1.0.0",ARGS.version_desc,temp_s3_url,product_id,product_name,region)
+            break
+            """
+
+    else:
+        product_id,product_version_id,product_version_name =create_product(ser_cat_clt_conn,product_name,temp_s3_url)
+        print "product {} created in region {} with version {}".format(product_name,region,product_version_name)
+        #TO associate the product with portfolio
+        attach_product_to_portfolio(ser_cat_clt_conn,product_id,portfolio_id)
+        print "product {} attached with portfolio {}".format(product_name,PORTFOLIO_NAME)
+
+
+
+
+if __name__ == "__main__":
+    ARGS = parse_arguments()
+    SUPPORT_EMAIL = "mohit@flux7.com"
```

Comment:
Do not hard-code these in the script. These should be taken as environment variables/cli args

Reply:
yes i am going to change it

Reply:
replace with cli args